### PR TITLE
Allow fireballs to be precise when shot

### PR DESCRIPTION
**Pull request by Pugzy**
Currently, fireballs have randomness when shot. Setting `precise` to true on the projectile fixes this.

`NMSHacks` needed as setting direction via API method call always has the random applied to it.

Signed-off-by: Pugzy <pugzy@mail.com>

### CODE REVIEW

**Comment by Electroid** [comment]
Should we just make this the default? Hard to imagine that the randomness is intended. (e.g. Not have a config value at all. If it's a fireball, just make it precise)

File: core/src/main/java/tc/oc/pgm/projectile/ProjectileMatchModule.java [hunk 1]
```diff
@@ -92,6 +94,10 @@ && isValidProjectileAction(event.getAction(), projectileDefinition.clickAction))
           projectile =
               player.launchProjectile(
                   projectileDefinition.projectile.asSubclass(Projectile.class), velocity);
+          if (projectile instanceof Fireball && projectileDefinition.precise) {
+            NMSHacks.setFireballDirection(
+                (Fireball) projectile, player.getEyeLocation().getDirection());
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/projectile/ProjectileMatchModule.java]:
```suggestion
                (Fireball) projectile, velocity);
```

**Comment by Cazadorian** [comment]
> Should we just make this the default? Hard to imagine that the randomness is intended. (e.g. Not have a config value at all. If it's a fireball, just make it precise)

I think it'd likely be best to have it be configurable as although in most cases mapmakers would want no randomness, there may be some cases where it is actually desirable to keep(i.e to limit the effective range of regular fireballs or to make gun style projectiles that use either type not effectively sniper rifles).

File: core/src/main/java/tc/oc/pgm/projectile/ProjectileModule.java [hunk 2]
```diff
@@ -69,6 +69,7 @@ public ProjectileModule parse(MapFactory factory, Logger logger, Document doc)
         Duration coolDown = XMLUtils.parseDuration(projectileElement.getAttribute("cooldown"));
         boolean throwable =
             XMLUtils.parseBoolean(projectileElement.getAttribute("throwable"), true);
+        boolean precise = XMLUtils.parseBoolean(projectileElement.getAttribute("precise"), false);
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/projectile/ProjectileModule.java]: Let's make `precise` default to `true`. Map makers can disable it.

Comment (calcastor) [reply core/src/main/java/tc/oc/pgm/projectile/ProjectileModule.java]: I think this is appropriate; there are little to no maps that use fireballs and "depend" on them being as janky as they tend to be.

**Review by Pablete1234 — APPROVED** [review]

**Review by Electroid — APPROVED** [review]

**Pull request merged**